mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,450 @@
+        Note: This depends on the `CheckpointHook` whose priority is more
+        than `WandbLogger`.
+
+    - Checkpoint Metadata: If `log_checkpoint_metadata` is True, every
+        checkpoint artifact will have a metadata associated with it.
+        The metadata contains the evaluation metrics computed on validation
+        data with that checkpoint along with the current epoch. If True, it
+        also marks the checkpoint version with the best evaluation metric with
+        a 'best' alias. You can choose the best checkpoint in the W&B Artifacts
+        UI using this.
+        Note: It depends on `EvalHook` whose priority is more than WandbLogger.
+
+    - Evaluation: At every evaluation interval, the `WandbLogger` logs the
+        model prediction as interactive W&B Tables. The number of samples
+        logged is given by `num_eval_images`. Please refer to
+        https://docs.wandb.ai/guides/data-vis to learn more about W&B Tables.
+        Currently, the `WandbLogger` logs the predicted bounding boxes along
+        with the ground truth at every evaluation interval.
+        Note: This depends on the `EvalHook` whose priority is more than
+        `WandbLogger`. Also note that the data is just logged once and
+        subsequent evaluation tables uses reference to the logged data to save
+        memory usage.
+
+    ```
+    Example:
+        log_config = dict(
+            interval=10,
+            hooks=[
+                dict(type='WandbLogger',
+                     wandb_init_kwargs={
+                         'entity': WANDB_ENTITY,
+                         'project': WANDB_PROJECT_NAME
+                     },
+                     logging_interval=10,
+                     log_checkpoint=True,
+                     log_checkpoint_metadata=True,
+                     num_eval_images=100)
+            ])
+    ```
+
+    Args:
+        wandb_init_kwargs (dict): A dict passed to wandb.init to initialize
+            a W&B run. Please refer to https://docs.wandb.ai/ref/python/init
+            for possible key-value pairs.
+        logging_interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        log_checkpoint_metadata (bool): Log the evaluation metrics computed
+            on the validation data with the checkpoint, along with current
+            epoch as a metadata to that checkpoint.
+            Default: True
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+    """
+
+    def __init__(self,
+                 wandb_init_kwargs=None,
+                 interval=10,
+                 log_checkpoint=False,
+                 log_checkpoint_metadata=False,
+                 num_eval_images=100):
```

Comment:
We can add **kwargs to allow users to custom the inputs of `mmcv.WandbLoggerHook`.

Reply:
Done. ✅